packages/ses/src/module-static-record.js

```diff
@@ -0,0 +1,44 @@
+import * as babel from '@agoric/babel-standalone';
+// We are using the above import form, and referring to its default export
+// explicitly below like babel.default, because the proper construct causes a
+// Rollup error.
+// This form:
+//   import babel from '@agoric/babel-standalone';
+// And this variant:
+//   import { default as babel } from '@agoric/babel-standalone';
+// Both produce:
+//   Error: 'default' is not exported by .../@agoric/babel-standalone/babel.js
+import { makeModuleAnalyzer } from '@agoric/transform-module';
+
+const analyze = makeModuleAnalyzer(babel.default);
+
+// moduleAnalyses are the private data of a ModuleStaticRecord.
+// We use moduleAnalyses in the loader/linker to look up
+// the analysis corresponding to any ModuleStaticRecord constructed by an
+// importHook.
+// This WeakMap is shared among modules in this package, but not exposed to the
+// public.
+export const moduleAnalyses = new WeakMap();
+
+// ModuleStaticRecord captures the effort of parsing and analyzing module text
+// so a cache of ModuleStaticRecords may be shared by multiple Compartments.
+export class ModuleStaticRecord {
+  constructor(text) {
+    const analysis = analyze({ string: text });
+    moduleAnalyses.set(this, analysis);
```

Comment:
Since you're exporting the `moduleAnalysis` weakmap, you should freeze `this` before placing it in the weakmap. In this case we know that adding it as a key to an exported weakmap doesn't give access anyway. But it makes the code harder to analyze for security properties.

Is `analysis` frozen?

Reply:
The `analysis` is not frozen. The code in subsequent reviews will access the `analysis` from the weak map, but does not modify it.

Reply:
Any reason not to freeze it and its contents?

Reply:
`analysis` is freezable. We can freeze it here, at its source (for better assurance that it will be completely frozen regardless of API drift, or harden it. Any recommendation?

Reply:
Freeze it at its source. If it doesn't create a layering problem, yes, `harden` it.

Reply:
https://github.com/Agoric/SES-shim/pull/280

Reply:
resolving